cherrypy/lib/auth_digest.py

```diff
@@ -142,15 +141,12 @@ def H(s):
     return md5_hex(s)
 
 
-def _try_decode_map_values(param_map, charset):
+def _try_decode_header(header, charset):
     global FALLBACK_CHARSET
 
     for enc in (charset, FALLBACK_CHARSET):
         try:
-            return {
-                k: tonative(v, enc)
-                for k, v in param_map.items()
-            }
+            return tonative(ntob(tonative(header, 'latin1'), 'latin1'), enc)
```

Comment:
This code is clearly not optimal or intuitive... and int introduces two additional deprecated calls. I want to clean this up.

Reply:
Yeah, I tried that. But it's a can of worms, there's no many moving parts. Let's do refactoring separately from the bug fix, please.